File: Java/src/com/leetcode/Medium/SimplifyPath.java
package com.leetcode.Medium;

import java.util.*;

// 71. Simplify Path
public class SimplifyPath {
    public String simplifyPath(String path) {
        Stack<String> stack = new Stack<>();
        String[] dirs = path.split("/");
        for (String dir : dirs) {
            if (!stack.empty() && dir.equals(".."))
                stack.pop();
            else if (!dir.equals(".") && !dir.equals("") && !dir.equals(".."))
                stack.push(dir);
        }
        List<String> list = new ArrayList<>(stack);
        return "/" + String.join("/", list);
    }
}

File: JavaScript/Medium/SimplifyPath.js
/**
 * 71. Simplify Path
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function (path) {
  const stack = [];
  const dirs = path.split("/");
  for (const dir of dirs) {
    if (stack.length && dir === "..") {
      stack.pop();
    } else if (dir !== "." && dir !== "" && dir !== "..") {
      stack.push(dir);
    }
  }

  const list = [...stack];
  return "/" + list.join("/");
};
